notification/job.go

```diff
@@ -0,0 +1,43 @@
+package notification
+
+import (
+	"fmt"
+
+	"github.com/flanksource/duty/context"
+	"github.com/flanksource/duty/job"
+	"github.com/flanksource/duty/models"
+	"gorm.io/gorm/clause"
+)
+
+func ProcessPendingNotificationsJob(ctx context.Context) *job.Job {
+	return &job.Job{
+		Name:       "ProcessPendingNotifications",
+		Retention:  job.RetentionFailed,
+		JobHistory: true,
+		RunNow:     true,
+		Context:    ctx,
+		Singleton:  false,
+		Schedule:   "@every 30s",
+		Fn: func(ctx job.JobRuntime) error {
+			tx := ctx.DB().Begin()
+			if tx.Error != nil {
+				return nil
+			}
+			defer tx.Rollback()
+
+			var pending []models.NotificationSendHistory
+			if err := tx.Clauses(clause.Locking{Strength: clause.LockingStrengthUpdate, Options: clause.LockingOptionsSkipLocked}).
+				Where("status = ?", models.NotificationStatusPending).
+				Where("delay IS NULL OR created_at + (delay * INTERVAL '1 second' / 1000000000)  <= NOW()").
+				Order("created_at + (delay * INTERVAL '1 second' / 1000000000)"). // smallest effective send time first
+				Limit(1).                                                         // one at a time; as one notification failure shouldn't affect a previous successful one
```

Comment:
This will rate limit notifications to once per 30s, this should probably be a continously running background goroutine that sleeps for 1m on empty